tests/python/relay/test_dense.py

```diff
@@ -0,0 +1,49 @@
+import tvm
+from tvm import relay
+from tvm.testing import assert_allclose
+import numpy as np
+from tvm.ir.instrument import pass_instrument
+
+
+def _test_accuracy(input_values, output_values, build_mod):
+
+    dev = tvm.cpu(0)
+
+    input_buf = tvm.nd.array(input_values, device=dev)
+    rt = tvm.contrib.graph_executor.GraphModule(build_mod["default"](dev))
+    rt.set_input("data", input_buf)
+    rt.run()
+    out = rt.get_output(0)
+
+    tvm.testing.assert_allclose(out.numpy(), output_values)
+
+
+# Define input shape and data type
+data_size = (64, 64)
+data_shape = data_size  # Input shape
+data_type = "float32"  # Data type
+weight_shape = data_size
+
+# Create Relay input variable
+d = relay.var("data", shape=data_shape, dtype=data_type)
+w1 = np.ones(weight_shape, dtype=data_type)
+w = relay.const(w1)
+
+# Create Relay dense layer
+y = relay.nn.dense(d, w)
+
+# Create Relay module
+mod = tvm.IRModule()
+
+# Define a Relay function with the dense layer
+mod["main"] = relay.Function([d], y)
+
+# Compile the Relay module
+target = "llvm -mtriple=aarch64-linux-gnu -device=arm_cpu -mattr=+v8.2a,+neon"  # Example target, you can change this to your desired target
+lib = relay.build(mod, target=target, params=None)
```

Comment:
To differentiate this Relay test from the TOPI one, I think it would be good to make sure the AlterOpLayout pass runs, either by calling it manually as you did [here](https://github.com/apache/tvm/pull/17091/files#diff-1acd41854c82cb9d55e9e17c34c5f972e98a6a6c517dcf6f448d39acef293278R1500), or by adding a pass context with `opt_level=3` like [here](https://github.com/apache/tvm/blob/main/tests/python/relay/strategy/arm_cpu/test_dense.py#L146-L147).  

That way we're also testing the the AlterOpLayout pass has not made the computation incorrect.